src/battle_ai_switch_items.c

```diff
@@ -897,37 +1079,774 @@ static u32 GetBestMonDmg(struct Pokemon *party, int firstId, int lastId, u8 inva
+        && defType1 != TYPE_FLYING && defType2 != TYPE_FLYING
+        && ability != ABILITY_LEVITATE && heldItemEffect != HOLD_EFFECT_AIR_BALLOON)
+        || (heldItemEffect == HOLD_EFFECT_IRON_BALL || (gFieldStatuses & STATUS_FIELD_GRAVITY))))
+    {
+        if (tSpikesLayers == 1)
+        {
+            AI_DATA->switchinCandidate.battlemon.status1 = STATUS1_POISON; // Assign "hypothetical" status to the switchin candidate so we can get the damage it would take from TSpikes
+            AI_DATA->switchinCandidate.hypotheticalStatus = TRUE;
+        }
+        if (tSpikesLayers == 2)
+        {
+            AI_DATA->switchinCandidate.battlemon.status1 = STATUS1_TOXIC_POISON; // Assign "hypothetical" status to the switchin candidate so we can get the damage it would take from TSpikes
+            AI_DATA->switchinCandidate.battlemon.status1 += STATUS1_TOXIC_TURN(1);
+            AI_DATA->switchinCandidate.hypotheticalStatus = TRUE;
+        }
+    }
+    return statusDamage;
+}
+
+// Gets number of hits to KO factoring in hazards, healing held items, status, and weather
+static u32 GetSwitchinHitsToKO(s32 damageTaken, u32 battler)
+{
+    u32 startingHP = AI_DATA->switchinCandidate.battlemon.hp - GetSwitchinHazardsDamage(battler, AI_DATA->switchinCandidate.battlemon);
+    s32 weatherImpact = GetSwitchinWeatherImpact(); // Signed to handle both damage and healing in the same value
+    u32 recurringDamage = GetSwitchinRecurringDamage();
+    u32 recurringHealing = GetSwitchinRecurringHealing();
+    u32 statusDamage = GetSwitchinStatusDamage(battler);
+    u32 hitsToKO = 0, singleUseItemHeal = 0;
+    u16 maxHP = AI_DATA->switchinCandidate.battlemon.maxHP, item = AI_DATA->switchinCandidate.battlemon.item;
+    u8 weatherDuration = gWishFutureKnock.weatherDuration;
+    u32 opposingBattler = GetBattlerAtPosition(BATTLE_OPPOSITE(GetBattlerPosition(battler)));
+    bool8 usedSingleUseHealingItem = FALSE;
+    s32 currentHP = startingHP;
+
+    // No damage being dealt
+    if (damageTaken + statusDamage + recurringDamage == 0)
+        return startingHP;
+
+    // Mon fainted to hazards
+    if (startingHP == 0)
+        return 1;
+
+    // Find hits to KO
+    while (currentHP > 0)
+    {
+        // Remove weather damage when it would run out
+        if (weatherImpact != 0 && weatherDuration == 0)
+            weatherImpact = 0;
+
+        // Take attack damage for the turn
+        currentHP = currentHP - damageTaken;
+
+        // If mon is still alive, apply weather impact first, as it might KO the mon before it can heal with its item (order is weather -> item -> status)
+        if (currentHP != 0)
+            currentHP = currentHP + weatherImpact;
+
+        // Check if we're at a single use healing item threshold
+        if (AI_DATA->switchinCandidate.battlemon.ability != ABILITY_KLUTZ && usedSingleUseHealingItem == FALSE)
+        {
+            if (currentHP < maxHP / 2)
+            {
+                if (item == ITEM_BERRY_JUICE)
```

Comment:
Instead of checking items by ID I think you should check if the item's hold effect is HOLD_EFFECT_RESTORE_HP or HOLD_EFFECT_RESTORE_PCT_HP and then the hp healed is in holdEffectParam. No need to check the I_SITRUS_BERRY_HEAL define then